src/Client.php

```diff
@@ -106,16 +106,16 @@ public function fetchToken(string $code): AccessToken
      *
      * @return \TestMonitor\Asana\AccessToken
      */
-    public function refreshToken(): AccessToken
+    public function refreshToken(): ?AccessToken
     {
         if (empty($this->token)) {
-            throw new UnauthorizedException();
+            throw new UnauthorizedException('Invalid access token');
         }
 
         try {
             $accessToken = $this->dispatcher->refreshAccessToken();
-        } catch (Exception $e) {
-            throw new InvalidTokenException($e->getMessage());
+        } catch (Exception $exception) {
+            throw new InvalidTokenException($exception->getMessage(), $exception->getCode(), $exception);
```

Comment:
We throw Unauthorized in the other clients?

Reply:
Changed 👍